### ISSUE TITLE
RenderPass::SetSrgsForDispatchRays() is missing

### ISSUE THREAD

**Issue by kh-huawei** [o3de/o3de#18275]
The `RayTracingPass` has to implement a custom handling for binding Shader-SRGs that ignores calls to the `BindSrg()` function.

Mostly because the functions `RenderPass::SetSrgsForDispatchRays()` and the subsequent (vendor-specific) `CommandList::SetShaderResourceGroupForDispatch()` are missing.

**Describe the solution you'd like**
The `RayTracingPass` behaves compareable to the `ComputePass`

**Comment by kh-huawei** [comment]
Various comments in the code also refer to [GFX TODO][ATOM-15610] Add RenderPass::SetSrgsForRayTracingDispatch

**Comment by moudgils** [comment]
I am assuming that this PR (https://github.com/o3de/o3de/pull/18255) addresses this issue?

**Comment by kh-huawei** [comment]
No, that PR only changes the RayTracingPass. This issue arose from that, because there is still some refactoring needed in the Commandlist (mostly moving logic from `CommandList::Submit()` to `CommitShaderResource()`) to have Raytracing SRGs work similar as other passes.
